File: StandarDeviasiSample/StandarDeviasiSample.java
import java.util.Scanner;

public class StandarDeviasiSample {
    public static void main(String[] args) {
        
        // Cetak judul
        System.out.println("========== PROGRAM STANDAR DEVIASI SAMPLE ==========");
        System.out.println("====================================================");

        // Buat objek scanner
        Scanner scan = new Scanner(System.in);

        // Terima input size array
        System.out.print("Banyak data: ");
        int size = scan.nextInt();

        // Buat array dengan size yang diinput
        double dataArray[] = new double[size];

        // Isikan dataArray
        for(int i = 0; i < size; i++) {
            System.out.print("Masukkan data ke-" + (i + 1) + ": ");
            dataArray[i] = scan.nextDouble();
        }

        // Tutup scanner
        scan.close();

        // Jumlahkan isi dataArray
        double sum = 0;
        for(int i = 0; i < size; i++) {
            sum += dataArray[i];
        }

        // Rata-rata array
        double average = sum / size;


        // Perhitungan pembilang
        double kurang, kuadrat = 0;
        for(int i = 0; i < size; i++) {
            kurang = dataArray[i] - average;
            kuadrat += (kurang * kurang);
        }

        // Perhitungan deviasi
        double deviasi = Math.sqrt(kuadrat / (size - 1));

        // Cetak hasil
        System.out.format("Standar deviasi sample: %.2f", deviasi);
    }
}

File: StandarDeviasiSample/StandarDeviasiSample.c
#include <stdio.h>
#include <math.h>

int main() {
    printf("========== STANDAR DEVIASI SAMPLE ==========\n");
    printf("============================================\n");

    int size;
    printf("Banyak data: ");
    scanf("%d", &size);

    int dataArray[size];

    for(int i = 0; i < size; i++) {
        printf("Masukkan data ke-%d: ", size);
        scanf("%lf", dataArray[i]);
    }

    double sum;
    for(int i = 0; i < size; i++) {
        sum += dataArray[i];
    }

    double average = sum / size;


    double kurang, kuadrat;
    for(int i = 1; i < size; i++) {
        kurang = dataArray[i] - average;
        kuadrat += (kurang * kurang);
    }

    double deviasi = sqrt(kuadrat / (size - 1));

    printf("Standar deviasi sample: %.2f\n", deviasi);

    return 0;
}